salt/fileserver/roots.py

```diff
@@ -386,7 +387,7 @@ def _translate_sep(path):
 
         for path in __opts__['file_roots'][load['saltenv']]:
             for root, dirs, files in os.walk(
-                    path,
+                    salt.utils.stringutils.to_unicode(path),
```

Comment:
The reason for using to_unicode here is because the path being passed in may not be unicode, but unicode characters in a file under that path will cause an error. But if you pass in a unicode string as the path, it forces "all" paths in the walk to be unicode, which prevents that UnicodeDecodeError.